Sources/Skylab/SkylabClient.swift

```diff
@@ -87,6 +87,14 @@ public class DefaultSkylabClient : SkylabClient {
             let session = URLSession.shared
 
             let userContext = self.addContext(user:self.user)
+            
+            let userId = userContext["user_id"]
+            let deviceId = userContext["device_id"]
+            if userId == nil && deviceId == nil {
+                print("[Skylab] WARN: user id and device id are nil; no variants fetched")
```

Comment:
should we still call the completion handler to make sure whatever should run still runs?

Reply:
Yep, good point. Im going to remove the short circuit all together.